src/utils/parseQuery.ts

```diff
@@ -0,0 +1,41 @@
+import { NextApiRequest } from 'next';
+import { GetServerSidePropsContext } from 'next';
+
+export interface ParsedQuery {
+  templateName: string;
+  parameters: any;
+  viewport: { width: number; height: number };
+}
+
+export function parseQuery(req: NextApiRequest | GetServerSidePropsContext): ParsedQuery {
+  const isApiRequest = (req: any): req is NextApiRequest => 'query' in req;
+
+  const query = isApiRequest(req) ? req.query : req.query;
+
+  const { templateName, width, height, parameters } = query;
```

Comment:
Looks like outdated code
```suggestion
  const { templateName, width, height, parameters } = req.query;
```

Reply:
updated the code